File: app/src/main/java/example/com/archtest/api/GitHubApi.java
package example.com.archtest.api;

import java.util.List;

import example.com.archtest.data.RepoInfo;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

import static example.com.archtest.Constants.API.REPOS_LIST_URL;

interface GitHubApi {

	@GET(REPOS_LIST_URL)
	public Call<List<RepoInfo>> getWeatherInfo(@Query("page") int pageNum, @Query("per_page") int itemsPerPage);

}

File: app/src/main/java/example/com/archtest/api/GitHubApi.kt
package example.com.archtest.api

import example.com.archtest.Constants.API.REPOS_LIST_URL
import example.com.archtest.data.RepoInfo
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface GitHubApi {
    @GET(REPOS_LIST_URL)
    abstract fun getWeatherInfo(@Query("page") pageNum: Int, @Query("per_page") itemsPerPage: Int): Call<List<RepoInfo>>

}